pkg/providers/ack/ack.go

```diff
@@ -35,31 +37,49 @@ import (
 
 type Provider interface {
 	GetNodeRegisterScript(context.Context, map[string]string, *v1alpha1.KubeletConfiguration) (string, error)
+	GetClusterCNI(context.Context) (string, error)
+	LivenessProbe(*http.Request) error
 }
 
 type DefaultProvider struct {
 	clusterID string
 	ackClient *ackclient.Client
+
+	muClusterCNI sync.RWMutex
+	clusterCNI   string
 }
 
 func NewDefaultProvider(clusterID string, ackClient *ackclient.Client) *DefaultProvider {
 	return &DefaultProvider{
-		clusterID: clusterID,
-		ackClient: ackClient,
+		clusterID:  clusterID,
+		ackClient:  ackClient,
+		clusterCNI: "",
 	}
 }
 
-func (p *DefaultProvider) GetClusterCNI(ctx context.Context) (string, error) {
+func (p *DefaultProvider) LivenessProbe(_ *http.Request) error {
+	p.muClusterCNI.Lock()
+	//nolint: staticcheck
+	p.muClusterCNI.Unlock()
```

Comment:
Is there any write operation here? Maybe it doesn't need the lock and unlock operation if just a read operation.

Reply:
It's used to check whether it is in dead-lock state. aws provider does this.

Reply:
Gotcha.